var/spack/repos/builtin/packages/py-pyomo/package.py

```diff
@@ -94,15 +95,17 @@ class PyPyomo(PythonPackage):
     depends_on("py-sphinx-toolbox@2.16:", when="@6.7.1:+docs", type=("run"))
     depends_on("py-sphinx-jinja2-compat@0.1.1:", when="@6.7.1:+docs", type=("run"))
     depends_on("py-enum-tools", when="@6.7.1:+docs", type=("run"))
-    depends_on("py-numpy", when="@6.1:+docs", type=("run"))
+    # Pyomo does not support NumPy2 (May 9, 2024)
+    depends_on("py-numpy@1", when="@6.1:+docs", type=("run"))
     depends_on("py-scipy", when="@6.4.2:+docs", type=("run"))
 
     # when optional is requested
     depends_on("py-dill", when="@6.1:+optional", type=("run"))
```

Comment:
Optional, for future reference. For clarity and grouping, you can bunch this all up with:
```suggestion
    with when("@6.1:+optional"):
        depends_on("py-dill", type=("run"))
```

Reply:
(Marking resolved so the auto-merge will go through.)